--- ../src/main/java/com/example/recorder/TradeRecorder.java	2025-12-05 10:00:37
+++ o11y--course--field--200-otel-logs--main/_challenges/02-otlp/TradeRecorder.java	2025-12-05 10:00:37
@@ -35,7 +35,7 @@
 
         Trade savedTrade = tradeRepo.save(trade);
 
-        log.atInfo().log("trade committed for " + trade.customerId);
+        log.atInfo().addKeyValue(Main.ATTRIBUTE_PREFIX + ".gc_time", utilities.getGarbageCollectorDeltaTime()).log("trade committed for " + trade.customerId);
 
         return savedTrade;
     }
