--- ../app.py	2025-09-22 21:10:51
+++ o11y--course--field--200-otel-logs--main/_challenges/02-otlp/app.py	2025-09-21 22:31:31
@@ -18,10 +18,6 @@
     MeterProvider,
 )
 
-from opentelemetry import _logs as logs
-from opentelemetry.processor.logrecord.baggage import BaggageLogRecordProcessor
-from opentelemetry.processor.baggage import BaggageSpanProcessor, ALLOW_ALL_BAGGAGE_KEYS
-
 ATTRIBUTE_PREFIX = "com.example"
 
 app = Flask(__name__)
@@ -43,7 +39,6 @@
 
     if 'OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED' in os.environ:
         print("enable otel logging")
-        logs.get_logger_provider().add_log_record_processor(BaggageLogRecordProcessor(ALLOW_ALL_BAGGAGE_KEYS))        
         root_logger = logging.getLogger()
         for handler in root_logger.handlers:
             if isinstance(handler, logging.StreamHandler):
@@ -88,6 +83,8 @@
     set_attribute_and_baggage(f"{ATTRIBUTE_PREFIX}.customer_id", customer_id)
 
     subscription = params.get('subscription', None)
+    if subscription is not None:
+        set_attribute_and_baggage(f"{ATTRIBUTE_PREFIX}.subscription", subscription)
 
     day_of_week = params.get('day_of_week', None)
     if day_of_week is None:
